packages/cli/src/workflows/workflow.service.ts

```diff
@@ -232,8 +232,8 @@ export class WorkflowService {
 
 		const newState = updatedWorkflow.active;
 
-		if (this.orchestrationService.isMultiMainSetupEnabled && oldState !== newState) {
-			await this.orchestrationService.publish('workflowActiveStateChanged', {
+		if (this.orchestrationService.isMultiMainSetupEnabled && this.orchestrationService.isFollower) {
```

Comment:
The initial idea was to have the follower inform the leader **if and only if** a trigger or poller was updated, but this turned out to be impractical:
- Currently there is no clean reliable way to tell apart non-HTTP-based triggers from HTTP-based triggers, as the `webhook()` or `trigger()` or `poller` methods are not available in the node object in the updated workflow and would require further querying. 
- Even in order to query, we'd have to inspect the workflow pre- and post-update to know exactly which node(s) were updated.
- If the update edited both a non-HTTP-based trigger and an HTTP-based trigger, the follower would need to handle the latter and inform about the former, and leader would need to know, in this edge case, not to activate the HTTP-based trigger, because it was already taken care of by the follower.

This would increase complexity rather than reducing it. Therefore I decided to have only the leader be responsible for managing webhooks, triggers and pollers - and to have followers only ever proxy to the leader. This simplifies the orchestration service methods and makes it easier to reason about all the flows, at the risk of the leader crashing when it is meant to react to a follower message. Open for discussion.

Reply:
Ok based on this comment and mine above, we might have to adjust 1 thing: the deregistration of webhooks should be handled by the node receiving the `patch` request, but the activation of all triggers is handled by the current leader.

Either this or we proxy the whole update process to the current leader (i.e. the http payload so that the current leader is responsible for running the entire process) but I think this is also too complicated.

We should just disable webhooks on the current node and forward the entire activation and all changes to the current leader, wdyt?